config/intel-flags

```diff
@@ -154,6 +157,16 @@ if test "X-icc" = "X-$cc_vendor"; then
         H5_CFLAGS="$H5_CFLAGS $(load_intel_arguments 15)"
     fi
 
+    # intel >= 18
+    if test $cc_vers_major -ge 18; then
+        H5_CFLAGS="$H5_CFLAGS $(load_intel_arguments 18)"
+    fi
+
+    # intel <= 19
```

Comment:
Effect is to add -Wcheck for intel version 19 and lower?  This is the reverse of the application of most of the compiler flags.  It might be worth adding a comment explaining why this one is different.